### ISSUE TITLE
set Alarm

### ISSUE THREAD

**Issue by milymat** [jishi/node-sonos-http-api#462]
there is no possibility to add an alarm to your sonos speaker.

Could u please implement it.

**Comment by jishi** [comment]
Adding an alarm is a fairly complex task, you need to reference time, volume and a sonos playlist, as well as specifying weekdays when it should be triggered. 

It's not a good fit for the GET based url scheme I'm using, so it's dependant on me extending the protocol to include support for more complex POST calls. Leaving this around so people can vote for it though. 

How would you use it? Why is the official app inconvenient for this?

**Comment by milymat** [comment]
Thx for your answer.

I would use it to set the alarm via some home automation software. So the speaker are independent and if the home automation software crashed the alarm still works.

My suggestion: You could use something like:     http://...//Alarm/HH/MM/PLAYLIST to set a simple alarm once

Check out these implementation, maybe I could help:
https://github.com/bencevans/node-sonos/blob/master/lib/services/AlarmClock.js

**Comment by Burnett2k** [comment]
I created my own alarn system using the npm package node-scheduler. It is pretty easy to set up a schedule and then run whatever you code you want when the timer hits. https://www.npmjs.com/package/node-schedule

**Comment by jcacavias** [comment]
I'm interesting in seeing this functionality.  I'm working on a private voice assistant and being able to call this rest api to set an alarm would be ideal.

**Comment by Blockmove** [comment]
I'm also very interested in this feature. IT's the only thing i'm missing.
Sometimes I have shift work with different start times.
I would be very usefull to set the sonos alarm clock with my homeautomation.

**Comment by Kaplonx** [comment]
I'm also interested in this functionality. By allowing to get/set/delete alarms using the API similar to Philips Hue (https://developers.meethue.com/develop/hue-api/3-schedules-api/), alarms set by the API would be reflected within the own Sonos App as well and modified from there also.

This would be very useful for a scenario where a user would like to set alarms for different devices from a central location.

**Comment by honato11** [comment]
Also very interested in this feature (getAlarm, setAlarm, deleteAlarm).

**Comment by vjooris** [comment]
One inconvenient of Sono, is that you have to be on site to manage them, and the alarms.
Should it be possible to develop an interrogation returning a son with all alarms set ?
And, as I do understand is is quite complex to create a new alarm, we could just modify one existing §volume, active, day(s), time) ?
This should be very nice, so we could set this up remotely.

**Comment by 4ch1m** [comment]
@jishi: Would it be possible to at least add a simple command to the existing API that lets one enable/disable existing alarms set up in the Sonos system?

Something like:
* `/[Room name]/alarm/[time]/[state]`
e.g. `/Bedroom/alarm/06-30/off`

And similar to the "sayall"-endpoint:
* `/alarmall/[time]/[state]`
e.g. `/alarmall/08-00/on`

I would love to be able to use a handy IoT-button on my nightstand to quickly turn on/off my alarms. :smile: 

Thanks a bunch for the project anyway!

**Comment by matthewchubb** [comment]
Yes, please add this feature so that I can trigger my phone alarm, hue lights and Sonos music all with one command dependent on the random time I need to wake up the following morning! Thanks

**Comment by jasonthefirs** [comment]
I would also love some alarm control. The speakers in my daughters’ rooms have weekday alarms set and I would love to automate a check of their school calendar to toggle the alarm off the night before a holiday and back on the night before school starts up again.
